File: src/main/java/dev/mazurkiewicz/quizer/question/infrastructure/config/QuestionConfig.java
package dev.mazurkiewicz.quizer.question.infrastructure.config;

import dev.mazurkiewicz.quizer.question.domain.port.QuestionRepository;
import dev.mazurkiewicz.quizer.question.domain.port.QuestionService;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
class QuestionConfig {

    @Bean
    QuestionService questionService(final QuestionRepository repository) {
        return new QuestionService(repository);
    }
}

File: src/main/kotlin/io/mazurkiewicz/quizer/question/infrastructure/config/QuestionConfig.kt
package io.mazurkiewicz.quizer.question.infrastructure.config

import io.mazurkiewicz.quizer.question.domain.port.QuestionRepository
import io.mazurkiewicz.quizer.question.domain.port.QuestionService
import io.mazurkiewicz.quizer.quiz.domain.port.QuizTemplateService
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class QuestionConfig {

    @Bean
    fun questionService(repository: QuestionRepository, templateService: QuizTemplateService) = QuestionService(repository, templateService)
}